lib/model/store.dart

```diff
@@ -994,20 +1021,27 @@ class UpdateMachine {
     }
 
     final stopwatch = Stopwatch()..start();
-    final initialSnapshot = await _registerQueueWithRetry(connection,
-      stopAndThrowIfNoAccount: stopAndThrowIfNoAccount);
+    InitialSnapshot? initialSnapshot;
+    try {
+      initialSnapshot = await _registerQueueWithRetry(connection,
+        stopAndThrowIfNoAccount: stopAndThrowIfNoAccount);
+    } on _ServerVersionUnsupportedException catch (e) {
+      // `!` is OK because _registerQueueWithRetry would have thrown a
+      // not-_ServerVersionUnsupportedException if no account
+      if (!e.data.matchesAccount(globalStore.getAccount(accountId)!)) {
+        await globalStore.updateZulipVersionData(accountId, e.data);
```

Comment:
I guess since the caller is going to log the account out anyway, this update isn't really needed. (Realized that when reading tests and seeing there doesn't seem to be a test exercising this.)

Having already done the work to write it, though, may as well keep it. It's more correct in principle, and will simplify if in the future we have it not log out in this case.